### PR TITLE
Added Generalised Headtracker Components

### PR DESCRIPTION
**Pull request by ExtremelySunnyYK**
#### Context
This PR moves over the Generalised HeadTracker components from core `common/types` repo to the chainlink-relay repo. This allows chain specific repositories to make use of the generalised Headtracker interfaces. 

#### Changes
- Moved over files from `common/types`
- Added mock files

#### Note:
- It is likely that we would have to move over the concrete types and methods for head tracker services - listener, broadcaster, and tracker. That would be in a separate PR. 

Ticket: https://smartcontract-it.atlassian.net/jira/software/c/projects/BCI/boards/23?modal=detail&selectedIssue=BCI-1116&assignee=712020%3A94dec959-6ba0-490a-b977-351d09b34915

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,3 +8,7 @@ godoc:
 	go install golang.org/x/tools/cmd/godoc@latest
 	# http://localhost:6060/pkg/github.com/smartcontractkit/chainlink-relay/
 	godoc -http=:6060
+
+.PHONY: mockery
+mockery: $(mockery) ## Install mockery.
+	go install github.com/vektra/mockery/v2@v2.28.1
```
Comment (jmank88) [inline Makefile:14]: New patch version released a few days ago:
```suggestion
	go install github.com/vektra/mockery/v2@v2.28.2
```

Comment (ExtremelySunnyYK) [reply Makefile:14]: Thanks Jordan for the heads up! will update that

**Review by jmank88 — DISMISSED** [review]

**Review by prashantkumar1982 — COMMENTED** [review]
Is it possible to wait on this, till you finish all those changes in the other PR that I commented on?
We can move repos once all changes are fully finished.

**Comment by ExtremelySunnyYK** [comment]
> Is it possible to wait on this, till you finish all those changes in the other PR that I commented on?
We can move repos once all changes are fully finished.

This PR will be pending until the other issues are addressed.

**Comment by ExtremelySunnyYK** [comment]
All the changes have been moved to this repo, ready for review

**Comment by ExtremelySunnyYK** [comment]
Closed PR. Created another updated one 

https://github.com/smartcontractkit/chainlink-relay/pull/139

**Pull request closed without merging**